src/Reader.js

```diff
@@ -419,15 +412,22 @@ class Reader extends EventEmitter {
 					}
 				}
 
-				try {
-					this.pendingLoadAuthenticationKey[key] = this.loadAuthenticationKey(parseInt(keyNumber), key);
-					await this.pendingLoadAuthenticationKey[key];
-				} catch (err) {
-					throw new AuthenticationError('unable_to_load_key', 'Could not load authentication key into reader.', err);
-				}
+				// Store the authentication promise in case other blocks are in process of authentication
+				this.pendingLoadAuthenticationKey[key] = this.loadAuthenticationKey(parseInt(keyNumber), key);
 
 			}
 
+			try {
+				await this.pendingLoadAuthenticationKey[key];
```

Comment:
Suppose the following situation:
1. key is not in the storage (condition `!keyNumber` is met)
2. there is already an authentication process for this key (so the condition `!this.pendingLoadAuthenticationKey[key]` is **not** met)
3. then try-catch-finally block is executed
4. **the `keyNumber` remains `null`** – **that's the problem** 
_You should assign return value of `await this.pendingLoadAuthenticationKey[key]` to `keyNumber` (like in [my code here on line 402](https://github.com/pokusew/nfc-pcsc/blob/master/src/Reader.js#L402))._
5. when the command is constructed, `null` results in `0x00` (which is the same as for `0`), so it will work for one key, but the error could appear when using two keys – that's probably the reason why you didn't notice when you were testing it

Suggested fix: replace line 421 with the following code:
```javascript
keyNumber = await this.pendingLoadAuthenticationKey[key];
```

P.S.: _Yeah, in some cases, we know the keyNumber, but because of simplified conditions (much cleaner code), we need to reassign the keyNumber in order to handle the situation described above._